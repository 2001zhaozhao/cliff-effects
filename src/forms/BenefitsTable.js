// REACT COMPONENTS
import React from 'react';
import {
  Table,
  Header
} from 'semantic-ui-react';

// CUSTOM COMPONENTS
// Both the table and graph should just be added to a results page, but
// this will do for now
import ResultsGraph from '../resultsGraph';

// BENEFIT LOGIC
import { getSNAPBenefits } from '../programs/state/massachusetts/snap';
import { getHousingBenefit } from '../programs/state/massachusetts/housing';


const getSignSymbol = function ( num ) {
  if ( num > 0 ) { return '+'; }
  else if ( num < 0 ) { return '-'; }
  else { return ''; }
};  // End getSignSymbol()


const BenefitsTable = function ( props ) {

  var client        = props.client,
      currentClient = { ...client };
  currentClient.futureEarnedIncomeMonthly = currentClient.currentEarnedIncomeMonthly;

  var SNAPBenefitCurrent  = Math.round( getSNAPBenefits( currentClient ).benefitValue * 12 ),
      SNAPBenefitFuture   = Math.round( getSNAPBenefits( client ).benefitValue * 12 ),
      SNAPDiff            = SNAPBenefitFuture - SNAPBenefitCurrent,
      sec8BenefitCurrent  = Math.round( getHousingBenefit( currentClient ).benefitValue * 12 ),
      sec8BenefitFuture   = Math.round( getHousingBenefit( client ).benefitValue * 12 ),
      sec8Diff            = sec8BenefitFuture - sec8BenefitCurrent,
      totalBenefitCurrent = SNAPBenefitCurrent + sec8BenefitCurrent,
      totalBenefitFuture  = SNAPBenefitFuture + sec8BenefitFuture,
      totalDiff           = SNAPDiff + sec8Diff,
      incomeCurrent       = Math.round( client.currentEarnedIncomeMonthly * 12 ),
      incomeFuture        = Math.round( client.futureEarnedIncomeMonthly * 12 ),
      incomeDiff          = incomeFuture - incomeCurrent,
      netCurrent          = totalBenefitCurrent + incomeCurrent,
      netFuture           = totalBenefitFuture + incomeFuture,
      netDiff             = totalDiff + incomeDiff;

console.log(SNAPDiff);



const cellStyle = function(entry) {

  const   basicCellStyle    = {textAlign: 'right'}
        , columnHeaderStyle = {background: 'rgba(0, 128, 128, 1)',
                              color: 'white',
                              fontSize: '1.3em',
                              fontWeight: 900,
                              textAlign: 'center',
                              borderRadius: 'inherit',
                              letterSpacing: '0.02em'
                              }
        , rowHeaderStyle    = {fontSize: '1.2em',
                              fontWeight: 700,
                              textAlign: 'left'}
        , totalsRowStyle    = {borderTop: "2px solid rgba(0, 128, 128, 1)"};


    switch(entry) {
        case 'basicCellStyle': 
          return basicCellStyle; 
        case 'columnHeaderStyle':
          return columnHeaderStyle; 
        case 'rowHeaderStyle' :
          return  Object.assign(basicCellStyle, rowHeaderStyle);
        case 'totalsRowStyle':
          return {function(){
                    Object.assign(basicCellStyle, totalsRowStyle);
                    }
                  };
        default: 
          return basicCellStyle;
                  }
  };


const Cell = function(props) {

  return (
    <Table.Cell width={3} style={cellStyle(props.style)}> {props.value} </Table.Cell>
  );
};



const Row = function(props) {
  const numbers = props.numbers;
  const cell = numbers.map((D) =>
    <Cell key={D.index}
        value={D} />
  );

  return (
    <Table.Row>
      {cell}
    </Table.Row>
  );
}

const numbers = ['Benefit', 'Current Benefits', 'New Estimate', 'Difference'];
const numbers1 = ['SNAP', SNAPBenefitCurrent, SNAPBenefitFuture, Math.abs(SNAPDiff)];
const numbers2 = ['Section 8 Housing', sec8BenefitCurrent, sec8BenefitFuture, Math.abs(sec8Diff)];
const numbers3 = ['Total Benefits', totalBenefitCurrent, totalBenefitFuture, Math.abs(totalDiff)];
const numbers4 = ['Income', incomeCurrent, incomeFuture, Math.abs(incomeDiff)];
const numbers5 = ['Net Total', netCurrent, netFuture, Math.abs(netDiff)];




  return (
    <wrapper>


      <Table celled borderRadius='0.25em 0.25em 0 0'>
        <Table.Body >
            <Row numbers={numbers} />
            <Row numbers={numbers1} />
            <Row numbers={numbers2} />
            <Row numbers={numbers3} />
            <Row numbers={numbers4} />
            <Row numbers={numbers5} />
        </Table.Body>
      </Table>

{/*
      <Table celled>
      <Table.Body>
       <Table.Header>  Removed for formatting purposes temporarily}
          <Table.Row textAlign='center' style={columnHeaderStyle}>
            <Table.Cell width={3}>Benefit</Table.Cell>
            <Table.Cell width={3}>Current Benefits</Table.Cell>
            <Table.Cell width={3}>New Estimate</Table.Cell>
            <Table.Cell width={3}>Difference</Table.Cell>
        </Table.Row>
        </Table.Header>
      </Table.Body>
        <Table.Body>
            <Table.Row>
              <Table.Cell textAlign='left'><Header as='h4'>SNAP</Header></Table.Cell>
              <Table.Cell textAlign='right'>${SNAPBenefitCurrent} / year</Table.Cell>
              <Table.Cell textAlign='right'>${SNAPBenefitFuture} / year</Table.Cell>
              <Table.Cell textAlign='right'>{ getSignSymbol(SNAPDiff) } ${Math.abs(SNAPDiff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell textAlign='left'><Header as='h4'>Section 8 Housing</Header></Table.Cell>
              <Table.Cell textAlign='right'>${sec8BenefitCurrent} / year</Table.Cell>
              <Table.Cell textAlign='right'>${sec8BenefitFuture} / year</Table.Cell>
              <Table.Cell textAlign='right'>{ getSignSymbol(sec8Diff) } ${Math.abs(sec8Diff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='left'><Header as='h4'>Total Benefits</Header></Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${totalBenefitCurrent} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${totalBenefitFuture} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>{ getSignSymbol(totalDiff) } ${Math.abs(totalDiff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row>
              <Table.Cell textAlign='left'><Header as='h4'>Income</Header></Table.Cell>
              <Table.Cell textAlign='right'>${incomeCurrent} / year</Table.Cell>
              <Table.Cell textAlign='right'>${incomeFuture} / year</Table.Cell>
              <Table.Cell textAlign='right'>{ getSignSymbol(incomeDiff) } ${Math.abs(incomeDiff)} / year</Table.Cell>
            </Table.Row>
            <Table.Row style={{border: 'none'}}>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='left'><Header as='h4'>Net Total</Header></Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${netCurrent} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>${netFuture} / year</Table.Cell>
              <Table.Cell width={3} style={rowHeaderStyle} textAlign='right'>{ getSignSymbol(netDiff) } ${Math.abs(netDiff)} / year</Table.Cell>
            </Table.Row>
        </Table.Body>

      </Table>
    )*/}

      <br/>
      <br/>

      <ResultsGraph {...props}/>
    </wrapper>
  );

};  // End BenefitsTable(<>)


export {
	BenefitsTable
};
