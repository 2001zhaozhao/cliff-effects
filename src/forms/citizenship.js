// REACT COMPONENTS
import React from 'react';
import { Form, Radio } from 'semantic-ui-react';

// PROJECT COMPONENTS
import {
  FormPartsContainer
} from './formHelpers';


/** @todo description
* 
* @function
* @param {object} props
* @property {object} props.__ - explanation
* 
* @returns Component
*/
const CitizenshipContent = ({ setClientProperty, client }) => {

  return (
    <wrapper className={'field-aligner'}>
      <Form.Field>
        <Radio
          label='US Citizen / National'
          name='citizenshipStatus'
          value='citizen'
          checked={client.citizenshipStatus === 'citizen'}
          onChange={setClientProperty}
        />
      </Form.Field>
      <br/>
      <Form.Field>
        <Radio
          label='Lawfully present immigrant / AWSS'
          name='citizenshipStatus'
          value='immigrant'
          checked={client.citizenshipStatus === 'immigrant'}
          onChange={setClientProperty}
        />
      </Form.Field>
      <br/>
      <Form.Field>
        <Radio
          label="Don't Know"
          name='citizenshipStatus'
          value='unknown'
          checked={client.citizenshipStatus === 'unknown'}
          onChange={setClientProperty}
        />
      </Form.Field>
    </wrapper>
  );  // end return

};  // End CitizenshipContent()

/** @todo description
* 
* @function
* @param {object} props
* @property {object} props.__ - explanation
* 
* @returns Component
*/
// `props` is a cloned version of the original props. References broken.
const CitizenshipStep = function ( props ) {

  return (
    <Form className='citizenship-form'>
      <FormPartsContainer
        title     = {'Citizenship Status'}
        clarifier = {'Select your citizenship status.'}
        left      = {{name: 'Previous', func: props.previousStep}}
        right     = {{name: 'Next', func: props.nextStep}}>
          <CitizenshipContent setClientProperty={props.setClientProperty} client={props.client} />
      </FormPartsContainer>
    </Form>
  );

};  // End CitizenshipStep()

export { CitizenshipStep };
