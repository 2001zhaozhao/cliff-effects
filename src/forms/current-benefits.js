// REACT COMPONENTS
import React from 'react';
import { Form } from 'semantic-ui-react';

// PROJECT COMPONENTS
import {
  FormPartsContainer, MassiveToggle
} from './formHelpers';


/** @todo description
* 
* @todo Add "vertical list of options" creator that will create a list of fields using the `.field-aligner` class
* 
* @function
* @param {object} props
* @property {object} props.__ - explanation
* 
* @returns Component
*/
const CurrentBenefitsContent = (props) => {

  var client = props.pageState;

  return (
    <wrapper className={'field-aligner'}>
      <MassiveToggle
        label={{ children: client.hasSnap ? <strong>SNAP</strong> : 'SNAP' }}
        checked={client.hasSnap}
        onChange={props.storeChecked}
        name='hasSnap' />
      <br/>
      <MassiveToggle
        label={{ children: client.hasHousing ? <strong>Section 8 Housing</strong> : 'Section 8 Housing' }}
        checked={client.hasHousing}
        onChange={props.storeChecked}
        name='hasHousing' />
      <br/>
      <MassiveToggle
        label={{ children: client.hasMassHealth ? <strong>MassHealth</strong> : 'MassHealth' }}
        checked={client.hasMassHealth}
        onChange={props.storeChecked}
        name='hasMassHealth' />
    </wrapper>
  );  // end return

};  // End CurrentBenefitsContent()

/** @todo description
* 
* @function
* @param {object} props
* @property {object} props.__ - explanation
* 
* @returns Component
*/
// `props` is a cloned version of the original props. References broken.
const CurrentBenefitsStep = (props) => {

  return (
    <Form size='massive' className='household-size-form'>
      <FormPartsContainer
        title     = {'Current Benefits'}
        clarifier = {'Select the benefits you currently receive.'}
        right     = {{name: 'Next', func: props.nextStep}}>
          <CurrentBenefitsContent {...props} />
      </FormPartsContainer>
    </Form>
  );

};  // End CurrentBenefitsStep()

export { CurrentBenefitsStep };
