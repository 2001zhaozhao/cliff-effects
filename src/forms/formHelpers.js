// REACT COMPONENTS
import React from 'react';
import {
  // Generic Form stuff
  Header,
  Segment,
  Label,
  Divider,
  Icon,
} from 'semantic-ui-react';

// PROJECT COMPONENTS
import { BigButton } from './inputs';


// ========================================
// GENERIC COMPONENTS
// ========================================

/**
 * Link that opens new tab
 */
const ExternalLink = function ({ href, children }) {
  return (
    <a
      href={ href }
      target='_blank'>{children}
    </a>);
};


/** For styling spacing between elements when needed.
* @returns Component
*/
const SpaceHolder = function () {
  return (<div className = { `space-holder` } />);
};


/** The row containing the big buttons at the bottom of each
*     form section, such as 'Previous', 'Next', and 'New Client'.
* 
* @object buttonProps
* @property text {string} - Text to show on the button.
* @property onClick {function} - Optional function to run on button click.
*
* @function
* @param {object} props - One object for each button
* @property {buttonProps} props.left
* @property {buttonProps} props.middle
* @property {buttonProps} props.right
*
* @returns Component
*/
const FormBottomRow = function({ left, middle, right }) {

  var Left    = <SpaceHolder key = { 'left' } />,
      Middle  = <SpaceHolder key = { 'middle' } />,
      Right   = <SpaceHolder key = { 'right' } />;

  if (left) {
    Left = (
      <BigButton
        key = { 'left' }
        onClick = { left.onClick }>
        { left.text }
      </BigButton>
    );
  }

  // Considering having a non-button as a label
  // for the page. Not sure how to preserve the
  // same style.
  if (middle) {
    Middle = (<div key = { 'middle' }>{ middle.text }</div>);
  }

  if (right) {
    Right = (
      <BigButton
        key = { 'right' }
        onClick = { right.onClick }>
        { right.text }
      </BigButton>
    );
  }

  var children = [
    Left,
    // Needed for first form section where there's no
    // 'Prev' button.
    Middle,
    Right,
  ];

  /** @todo Move styles to CSS */
  return (
    <div
      className = { `form-section-bottom-row` }
      style     = {{ display: `flex`, justifyContent: `space-between` }}>
      { children }
    </div>
  );

}; // End <FormBottomRow>


/** Constructor for all the stuff that's supposed to go inside
* the Form Component. Does not include the `<Form>` Component
* as a container because it looks like that needs to be unique
* (the 'CurrentBenefitsStep' gives it `size='massive'`).
*
* @function
* @param {object} props
* @property {string} props.title - Text to go in the `h1` element.
* @property {string} props.clarifier - Text to go in the `h3`
* element, giving some description, instructions, or clarifications.
* @property {string} props.children - Component(s) to be inserted
* into the middle - a custom form section containing inputs, etc.
* @property {Object} props.next - the 'next form section' function
* @property {Object} props.prev - the 'previous form section' function
*
* @returns Component
*/
const FormPartsContainer = function({ title, clarifier, children, navData }) {
  return (
    <Segment
      padded='very'
      className="flex-item flex-column">
      <Segment
        basic={ true }
        className="flex-item">
        <Header
          as='h1'
          color='teal'
          textAlign='center'>
          { title }
        </Header>
        { !clarifier
          ? null
          : (
            <Header
              as='h3'
              textAlign='center'>
              { clarifier }
            </Header>
          )
        }

        { children }

      </Segment>

      <Divider />
      <FormBottomRow { ...navData } />

    </Segment>
  );
};  // End FormPartsContainer() Component


/** A clearer way than a ternary operator to have a possible
* subheader. Also, clearer for separate styling
*
* @function
* @param {object} props
* @property {object} props.children - Contents of this element
*
* @returns Component
*/
const FormSubheading = function (props) {

  if (!props.children) {
    return null;
  }

  return (
    <div
      className = { 'form-subheading' }
      style={{ display: 'block', textAlign: 'left' }}>
      { props.children }
    </div>
  );

};  // End FormSubheading{} Component


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const FormHeading = function ({ subheading, children }) {

  if (!children) {
    return null;
  }

  return (
    <div className={ 'form-heading' } >
      <div /> {/** div here to make sure header margin doesn\'t collapse */}
      <Header
        as='h3'
        style={{ display: 'inline-block' }}>
        { children }
      </Header>
      <FormSubheading>{subheading}</FormSubheading>
      <br />
    </div>
  );

};  // End FormHeading{} Component


// ========================================
// INPUT CONTAINER COMPONENTS
// ========================================

/** Adds an option for an 'invalid input' message to the right of the last element */
const InvalidMessage = function ({ validRow, message }) {

  var result = null;
  if (!validRow && message) {
    result = (
      <Label
        basic
        color='red'
        pointing="left">{message}
      </Label>
    );
  }

  return result;
};  // End <InvalidMessage>


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const ColumnHeading = function ({ type, colName, style, children }) {
  var classes = type + '-column cashflow-column header ' + colName;
  return (
    <Header
      as='h4'
      className={ classes }
      style={ style }
      color='teal'>{children}
    </Header>
  );
};  // End ColumnHeading()


/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const IntervalColumnHeadings = function ({ type }) {

  var columnTitle = type.toLowerCase().replace(/\b[a-z]/g, (letter) => {
        return letter.toUpperCase();
      }) + ' Type',
      styles      = { fontSize: '14px' };

  return (
    <div style={{ display: 'inline-block' }}>
      <ColumnHeading
        type={ type }
        colName='weekly'
        style={ styles }>Weekly
      </ColumnHeading>
      <ColumnHeading
        type={ type }
        colName='monthly'
        style={ styles }>Monthly
      </ColumnHeading>
      <ColumnHeading
        type={ type }
        colName='yearly'
        style={ styles }>Yearly
      </ColumnHeading>
      <ColumnHeading
        type={ type }
        colName={ type }
        style={ styles }
        columnTitle={ columnTitle }>{columnTitle}
      </ColumnHeading>
    </div>
  );

};  // End IntervalColumnHeadings{} Component


var AttentionArrow = function () {

  return (
    <span className={ 'attention-arrow' }>
      <Icon
        className = { 'attention-font' }
        fitted
        name      = { 'angle right' }
        size      = { 'big' } />
      <Icon
        className = { 'attention-font' }
        fitted
        name      = { 'angle right' }
        size      = { 'big' } />
    </span>
  );

};  // End <AttentionArrow>


/** @todo Separate into different files? */
export {
  ExternalLink, SpaceHolder,
  FormBottomRow,
  FormPartsContainer,
  FormSubheading, FormHeading,
  InvalidMessage,
  IntervalColumnHeadings, ColumnHeading,
  AttentionArrow,
};
