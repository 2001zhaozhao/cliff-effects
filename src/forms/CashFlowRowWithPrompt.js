import React from 'react';
import { Form } from 'semantic-ui-react';

import { CashFlowRow } from './formHelpers';

const NumberField = (props) => (
  <CashFlowRow {...props} type={'income'} />
);

/**
 * Prompt the user before presenting a CashFlowRow
 * 
 * @param {object} props
 * @param {string} props.generic - The key of the value being set.
 * @param {string} props.promptLabel - Label for initial prompt.
 * @param {string} props.children - Label for fields updating the value.
 * 
 * @extends React.Component
 * @see CashFlowRow
 */
class CashFlowRowWithPrompt extends React.Component {
  constructor(props) {
    super(props);
    
    const value = props.timeState[props.generic];
    this.state = {
      showField: value !== 0,
      storedValue: value
    }
  }

  handleChange = (evt, inputProps) => {
    if(inputProps.value === 'Yes') {
      this.showField(evt);
    } else {
      this.hideField(evt);
    }
  }

  hideField = evt => {
    const { generic, setClientProperty, timeState } = this.props;

    this.setState({
      showField: false,
      storedValue: timeState[generic]
    });

    setClientProperty(evt, {
      name: generic,
      value: 0
    });
  }

  showField = evt => {
    const { generic, setClientProperty } = this.props;
    const { storedValue } = this.state;

    setClientProperty(evt, {
      name: generic,
      value: storedValue
    });

    this.setState({
      showField: true
    });
  }

  render() {
    const { promptLabel, ...rest } = this.props;
    const { showField } = this.state;

    return (
      <div>
        <Form.Group inline>
          <label>{promptLabel}</label>
          <Form.Radio label={'Yes'} value={'Yes'} checked={showField} onChange={this.handleChange} />
          <Form.Radio label={'No'} value={'No'} checked={!showField} onChange={this.handleChange} />
        </Form.Group>
        
        {showField && <NumberField {...rest} />}
      </div>
    );
  }
}

export default CashFlowRowWithPrompt;
