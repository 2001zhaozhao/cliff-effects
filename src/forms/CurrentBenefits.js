// REACT COMPONENTS
import React from 'react';
import { Form } from 'semantic-ui-react';

// PROJECT COMPONENTS
import { FormPartsContainer, MassiveToggle, ControlledRadioYesNo } from './formHelpers';

// COMPONENT HELPER FUNCTIONS
import { getTimeSetter } from '../utils/getTimeSetter';


/** @todo description
*
* @todo Add "vertical list of options" creator that will create a list of fields using the `.field-aligner` class
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
const CurrentBenefitsContent = ({ current, setClientProperty }) => {

  let setChecked = function ( evnt, inputProps ) {
    inputProps.checked = inputProps.label === 'yes' ? true : false;
    var obj = { ...inputProps, value: inputProps.value };
    setClientProperty( evnt, obj );
  };  // End setChecked()


  return (
    <div className={'field-aligner'}>

      <ControlledRadioYesNo
        labelText = "Do you have Section 8 Housing ?"
        checked={current.hasHousing}
        onChange={setChecked}
        name="hasHousing"
      />

      <ControlledRadioYesNo
        labelText = "Do you have Snap?"
        checked={current.hasSnap}
        onChange={setChecked}
        name="hasSnap"
      />
    </div>
  );  // end return

};  // End CurrentBenefitsContent()

/** @todo description
*
* @function
* @param {object} props
* @property {object} props.__ - explanation
*
* @returns Component
*/
// `props` is a cloned version of the original props. References broken.
const CurrentBenefitsStep = (props) => {

  const setTimeProp = getTimeSetter( 'current', props.changeClient );

  return (
    <Form size='massive' className='household-size-form flex-item flex-column'>
      <FormPartsContainer
        title     = {'Current Benefits'}
        clarifier = {'Select the benefits you currently receive.'}
        right     = {{name: 'Next', func: props.nextStep}}>
          <CurrentBenefitsContent
          setClientProperty={setTimeProp}
          current={props.client.current} />
      </FormPartsContainer>

    </Form>
  );

};  // End CurrentBenefitsStep()

export { CurrentBenefitsStep };
