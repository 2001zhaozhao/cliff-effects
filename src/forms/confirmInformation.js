// REACT COMPONENTS
import React from 'react';
import { Header } from 'semantic-ui-react';

// PROJECT COMPONENTS
import { FormPartsContainer } from './formHelpers';

const ConfirmInformation = (props) => (
		<FormPartsContainer
			title     = 'Confirm Information'
			clarifier = ''
			left      = {{name: 'Previous', func: props.previousStep}}
			right     = {{name: 'Next', func: props.nextStep}}>
			  <div>
				<Header as='span' color='teal'>Current Benefits</Header> 
				{props.client.hasSnap ? "SNAP, ": ""}
				{props.client.hasHousing ? "Section 8, ": ""}
				{props.client.hasMassHealth ? "MassHealth": ""}<br/>
				<Header as='span' color='teal'>Household Size</Header> {props.client.householdSize}<br/>
				<Header as='span' color='teal'>Household Annual Income</Header> {props.client.annualIncome}<br/>
				<Header as='span' color='teal'>Citizenship Status</Header> {props.client.citizenshipStatus == "citizen" ? "US Citizen" : "Resident" }<br/>
				<Header as='span' color='teal'>MassHealth</Header> {props.client.hasMassHealth ? "Yes, has qualifying conditions" : "No qualifying conditions"  }
			  </div>
      </FormPartsContainer>
	  )

export { ConfirmInformation };