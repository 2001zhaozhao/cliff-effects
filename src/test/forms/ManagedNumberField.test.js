import React from 'react';
import { shallow } from 'enzyme';

import { ManagedNumberField } from '../../forms/formHelpers';

test('ManagedNumberField should match snapshot', () => {
  const wrapper = shallow(
    <ManagedNumberField
      format={ () => {} }
      value={ 0 } />
  );
  expect(wrapper).toMatchSnapshot();
});

test('should set focused state to true on focus if current value of FormInput is positive number', () => {
  const value = 10;
  const wrapper = shallow(
    <ManagedNumberField
      format={ () => {} }
      value={ value } />
  );
  wrapper.find('FormInput').simulate('focus', { target: { value }});
  expect(wrapper.state('focused')).toBe(true);
});

test('should set focused state to true on focus if current value of FormInput is 0', () => {
  const value = 0;
  const wrapper = shallow(
    <ManagedNumberField
      format={ () => {} }
      value={ value } />
  );
  wrapper.find('FormInput').simulate('focus', { target: { value }});
  expect(wrapper.state('focused')).toBe(true);
});

test('should blank focusedVal state before focus if current value of FormInput is 0', () => {
  const inputValue = 0;
  const wrapper = shallow(
    <ManagedNumberField
      format={ () => {} }
      value={ 0 } />
  );
  wrapper.find('FormInput').simulate('focus', { target: { value: inputValue }});
  expect(wrapper.state('focusedVal')).toBe('');
});

test('should set focused state to false on blur', () => {
  const value = 10;
  const wrapper = shallow(
    <ManagedNumberField
      format={ () => {} }
      onBlur={ () => {} }
      value={ value } />
  );
  wrapper.setState({ focused: true });
  wrapper.find('FormInput').simulate('blur', { target: { value }});
  expect(wrapper.state('focused')).toBe(false);
});
