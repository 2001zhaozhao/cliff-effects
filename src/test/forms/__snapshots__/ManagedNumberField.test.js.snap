// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ManagedNumberField should match snapshot 1`] = `
<FormInput
  as={[Function]}
  control={[Function]}
  error={false}
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
/>
`;
